xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -66,11 +66,19 @@ void CEngineStats::AddSamples(int samples, std::list<CActiveAEStream*> &streams)
 void CEngineStats::GetDelay(AEDelayStatus& status)
 {
   CSingleLock lock(m_lock);
+
+  bool isTrueHD = false;
+#if defined(TARGET_ANDROID)
```

Comment:
It doesn't seem to me that this is the correct way to handle this.

There are reports in the forum with the latest changes by @fritsch and "Kodi IEC packer" works fine.

The problem seems to be only with "Android IEC packer": 
https://forum.kodi.tv/showthread.php?tid=365127&pid=3068556#pid3068556

And for this I had offered a better solution:
https://github.com/thexai/xbmc/commit/b8caee8dc46479e2a46e1a0bef3762374372ea15

My reasoning is exposed here:
https://forum.kodi.tv/showthread.php?tid=365127&pid=3068870#pid3068870

Reply:
The RAW change by @thexai  is correct. The rest of the PR - is entirely not.

Reply:
Thanks. The only reason I have not opened a PR is because I have no way to test it and no one has given feedback yet.

Reply:
And RAW relied on that fact as well. We just counted frames and according to above formula the time added was always the same and static. Now we have a bit more dynamic there.

Btw. then the packer code was not changed, we now have twice the overhead (not negatively meant) that we send per frame (payload is half), but padding still happens. That means IEC sinks should work exactly as before.

For the RAW, maybe it's enough to just linearly adjust the "time per frame" (or frameset, so that it works as it did before?)

Reply:
**Kodi IEC path:**
```
DVDAudioCodecPassthrough ---(12 units)---> MAT packer ----(24 units)---> Kodi IEC packer ----(24 units)---> AVR
```

**RAW path:**
```
DVDAudioCodecPassthrough ---(12 units)---> ActiveESink ----(24 units RAW)---> Android IEC packer ----(24 units)---> AVR
```

Reply:
Yes. Exactly. So time for raw needs adjusting. Package wise the sink does not care as the frame number is just one always.